x-pack/plugins/lens/public/xy_visualization/xy_suggestions.ts

```diff
@@ -117,8 +117,9 @@ function getBucketMappings(table: TableSuggestion, currentState?: State) {
 
   if (
     currentXScaleType &&
-    // make sure histograms get mapped to x dimension even when changing current bucket/dimension mapping
-    (currentXScaleType === 'interval' || prioritizedBuckets[0].operation.scale !== 'interval')
+    // make sure date histograms get mapped to x dimension even when changing current bucket/dimension mapping
+    (prioritizedBuckets[currentXColumnIndex].operation.dataType === 'date' ||
+      prioritizedBuckets[0].operation.scale !== 'interval')
```

Comment:
I just made this change so that we could decouple the visualizations from specific knowledge of the source. I believe there is a less-coupled way to get the same results, such as by changing the way the datasource builds suggestions: https://github.com/elastic/kibana/pull/75703